python/bin/p1_lband_extract

```diff
@@ -0,0 +1,147 @@
+             "(see find_fusion_engine_log() and --log-base-dir)")
+
+    options = parser.parse_args()
+
+    read_index = not options.ignore_index
+    generate_index = not options.ignore_index
+
+    # Configure logging.
+    if options.verbose >= 1:
+        logging.basicConfig(level=logging.INFO, format='%(levelname)s - %(name)s:%(lineno)d - %(message)s',
+                            stream=sys.stdout)
+        if options.verbose == 1:
+            logging.getLogger('point_one.fusion_engine.parsers').setLevel(logging.DEBUG)
+        else:
+            logging.getLogger('point_one.fusion_engine.parsers').setLevel(
+                logging.getTraceLevel(depth=options.verbose - 1))
+    else:
+        logging.basicConfig(level=logging.INFO, format='%(message)s', stream=sys.stdout)
+
+    HighlightFormatter.install(color=True, standoff_level=logging.WARNING)
+    BrokenPipeStreamHandler.install()
+
+    # Locate the input file and set the output directory.
+    input_path, output_dir, log_id = locate_log(input_path=options.log, log_base_dir=options.log_base_dir, return_log_id=True, return_output_dir=True,
+                                                extract_fusion_engine_data=False)
+    if input_path is None:
+        # locate_log() will log an error.
+        sys.exit(1)
+
+    if options.output is not None:
+        output_dir = options.output
+
+    # Read through the data file, searching for valid LBandFrame messages to extract and store in
+    # 'output_dir/<prefix>.p1log'.
+    if options.prefix is not None:
+        prefix = options.prefix
+    elif log_id is not None:
+        prefix = 'lband'
+    else:
+        prefix = os.path.splitext(os.path.basename(input_path))[0]
+    output_path = os.path.join(output_dir, prefix + '.rtcm')
+
+    _logger.info("Processing input file '%s'." % input_path)
+
+    # Parse the time range.
+    time_range = TimeRange.parse(options.time, absolute=options.absolute_time)
+
+    if options.time is not None:
+        _logger.info('Non-P1 time messages requested and time range specified. Disabling index file.')
+        read_index = False
+
+    message_types = MessageType.LBAND_FRAME
+
+    # Process all data in the file.
+    reader = MixedLogReader(input_path, return_bytes=True, return_offset=True, show_progress=options.progress,
+                            ignore_index=not read_index, generate_index=generate_index,
+                            message_types=message_types, time_range=time_range)
+
+    if reader.generating_index() and (len(message_types) > 0 or options.time is not None):
+        _logger.info('Generating index file - processing complete file. This may take some time.')
+
+    total_messages = 0
+    bytes_decoded = 0
+
+    with open(output_path, 'wb') as fd:
+        try:
+            for header, message, data, offset_bytes in reader:
+                if isinstance(message, LBandFrameMessage):
```

Comment:
This is inefficient - it reads in the whole file. We should apply this as a filter to the reader before starting the loop. Then, when there's an index file, it will only read the L-band messages.